### DISCUSSION TITLE
Comparison with lsp-status.nvim

### DISCUSSION THREAD

**Discussion by donbex** [WhoIsSethDaniel/lualine-lsp-progress.nvim#3] [General]
First of all, thanks for this project. Since [lsp-status.nvim](https://github.com/nvim-lua/lsp-status.nvim) has been around for a while and appears to fulfil the same purpose, I was wondering if you could spend a moment to explain how it differs from `lualine-lsp-progress.nvim`.

I'm especially curious because `lsp-status.nvim` requires you to declare a capability and register a new handler in the lsp configuration, while it doesn't seem like `lualine-lsp-progress.nvim` does.

**Comment by WhoIsSethDaniel** [comment]
[lualine-lsp-progress.nvim](https://github.com/WhoIsSethDaniel/lualine-lsp-progress.nvim) is not really that new. It's a fork of [lualine-lsp-progress](https://github.com/arkav/lualine-lsp-progress) which was first started around 16 months ago (but hasn't seen any updates in close to a year -- and the author seems to have fallen off the face of the earth). This still makes it newer than `lsp-status.nvim` by about 8-10 months.

I had been using `lualine-lsp-progress` for at least a year before I started to see a few (small) problems and realized that it was unmaintained. It's only in the past month that I decided to start maintaining it.

I'm not *that* familiar with `lsp-status.nvim`. I know that I used it in the distant past. It was likely the first lsp status plugin that I used. I must have liked `lualine-lsp-progress` better since I've been using it for so long. 

You shouldn't have to set client capabilities as I understand it. The server will advertise that it will be sending progress reports and the client, by default, accepts them (I'll need to look at the lsp client code to verify this, but this is my recollection). I'm not certain about the handler question. `lualine-lsp-progress` simply [changed the default progress handler](https://github.com/arkav/lualine-lsp-progress/blob/56842d097245a08d77912edf5f2a69ba29f275d7/lua/lualine/components/lsp_progress.lua#L130). This has been the usual way of dealing with status messages up until recently. `lualine-lsp-progress.nvim` has been changed to use the `LspProgressUpdate` event instead of using the progress handler.
